tests/manage/pv_services/pvc_snapshot/test_small_files_pvc_cloned_status.py

```diff
@@ -0,0 +1,123 @@
+import logging
+import os
+import urllib.request
+import pytest
+
+from ocs_ci.ocs import constants
+from ocs_ci.ocs.ocp import OCP
+from ocs_ci.framework.testlib import (
+    E2ETest,
+    tier1,
+    polarion_id,
+    bugzilla,
+)
+
+log = logging.getLogger(__name__)
+
+
+@tier1
+@polarion_id("OCS-3946")
+class TestPvcClonedStatusOfSmallSizeFiles(E2ETest):
+    """
+    Tests to verify restored PVC status bound or not if
+    files size are smaller.
+    """
+
+    pvc_size = 1
+    pod_path = "/mnt"
+
+    @pytest.fixture()
+    def pvc(self, pvc_factory_class):
+        # Create a RWX Cephfs PVC
+        self.pvc_obj = pvc_factory_class(
+            interface=constants.CEPHFILESYSTEM,
+            access_mode=constants.ACCESS_MODE_RWX,
+            size=self.pvc_size,
+        )
+
+    @pytest.fixture()
+    def pod(self, pod_factory_class):
+        self.pod_obj = pod_factory_class(
+            pvc=self.pvc_obj,
+            status=constants.STATUS_RUNNING,
+            pod_dict_path=constants.PERF_POD_YAML,
+        )
+
+    @pytest.fixture()
+    def copy_files(self):
+        kernel_url = "https://cdn.kernel.org/pub/linux/kernel/v4.x/linux-4.4.tar.xz"
+        download_path = "tmp"
+        dir_path = os.path.join(os.getcwd(), download_path)
+        file_path = os.path.join(dir_path, "linux-4.4.tar.xz")
+        if not os.path.exists(dir_path):
+            os.makedirs(dir_path)
+        urllib.request.urlretrieve(kernel_url, file_path)
+
+        pod_name = self.pod_obj.name
+        ocp = OCP(namespace=self.pvc_obj.namespace)
+
+        rsh_cmd = f"cp {dir_path} {pod_name}:{self.pod_path}"
+        ocp.exec_oc_cmd(rsh_cmd)
+
+        rsh_cmd = f"tar xvf {self.pod_path}/tmp/linux-4.4.tar.xz -C {self.pod_path}/tmp"
+        self.pod_obj.exec_sh_cmd_on_pod(command=rsh_cmd)
+
+        rsh_cmd = f"ls {self.pod_path}/tmp/linux*|wc -l"
+        self.original_total_files = self.pod_obj.exec_sh_cmd_on_pod(command=rsh_cmd)
+        log.info(f"original_total_files: {self.original_total_files}")
+
+    @pytest.fixture()
+    def create_snapshot(self, snapshot_factory):
+        log.info(f"Creating snapshot from {self.pvc_obj.name}")
+        self.snapshot = snapshot_factory(
+            self.pvc_obj, snapshot_name=f"{self.pvc_obj.name}--snapshot"
+        )
+
+    @pytest.fixture()
+    def create_restore(self, snapshot_restore_factory):
+        log.info(f"Creating restore from snapshot {self.snapshot.name}")
+        self.pvc_restore = snapshot_restore_factory(
```

Comment:
Please change the order of the parameters to be as in snapshot_restore_factory().
It's not critical but nice to have